jfrog-cli/jfrog/buildinfo_test.go

```diff
@@ -38,6 +41,85 @@ func TestBuildAddDependenciesFromHomeDir(t *testing.T) {
 	cleanArtifactoryTest()
 }
 
+func TestBuildPromote(t *testing.T) {
+	initArtifactoryTest(t)
+	buildName, buildNumberA := "cli-test-build", "10"
+
+	// Upload files with buildName and buildNumber
+	specFileA, err := tests.CreateSpec(tests.SplitUploadSpecA)
+	if err != nil {
+		t.Error(err)
+	}
+	artifactoryCli.Exec("upload", "--spec="+specFileA, "--build-name="+buildName, "--build-number="+buildNumberA)
+	artifactoryCli.Exec("build-publish", buildName, buildNumberA)
+
+	key1 := "key"
+	value1 := "v1,v2"
+	key2 := "another"
+	value2 := "property"
+	artifactoryCli.Exec("build-promote", buildName, buildNumberA, tests.Repo2, fmt.Sprintf("--props=%s=%s;%s=%s", key1, value1, key2, value2))
+	buildInfo := inttestutils.GetBuildInfo(artifactoryDetails.Url, buildName, buildNumberA, t, artHttpDetails)
+	resultItems := getResultItemsFromArtifactory(tests.SearchRepo2, t)
+
+	if len(buildInfo.Modules[0].Artifacts) != len(resultItems) {
+		t.Error("Incorrect number of artifacts were uploaded, expected:", len(buildInfo.Modules[0].Artifacts), " Found:", len(resultItems))
+	}
+
+	propsMap := make(map[string]string)
+	propsMap["build.name"] = buildInfo.Name
+	propsMap["build.number"] = buildInfo.Number
+	propsMap[key1] = value1
+	propsMap[key2] = value2
+
+	validateArtifactsProperties(resultItems, t, propsMap)
+	cleanArtifactoryTest()
+}
+
+// Returns the artifacts result found by the provided spec
```

Comment:
// Returns the artifacts **result** found by the provided spec --> // Returns the artifacts found by the provided spec

Reply:
Done